### PR TITLE
remove realpath call in relative file name code path of dt_util_normalize_path()

### PR DESCRIPTION
**Pull request by UweOhse**
This is for #9734, and removes an inconsistency between the handling of absolute and relative file name arguments.

See https://github.com/darktable-org/darktable/issues/9734#issuecomment-1046087732 for the reasoning.

### CODE REVIEW

File: src/common/utility.c [hunk 1]
```diff
@@ -671,16 +671,8 @@ gchar *dt_util_normalize_path(const gchar *_input)
     char *current_dir = g_get_current_dir();
     char *tmp_filename = g_build_filename(current_dir, filename, NULL);
     g_free(filename);
-    filename = g_realpath(tmp_filename);
```
Comment (TurboGit) [inline src/common/utility.c:674]: Then we get the full path name resolving all symlinks.

File: src/common/utility.c [hunk 2]
```diff
@@ -671,16 +671,8 @@ gchar *dt_util_normalize_path(const gchar *_input)
...
-    if(filename == NULL)
-    {
-      g_free(current_dir);
-      g_free(tmp_filename);
-      g_free(filename);
-      return NULL;
-    }
+    filename = tmp_filename;
```
Comment (TurboGit) [inline src/common/utility.c:674]: Now you set filename to the full path name without resolving the symlinks.

This seems quite wrong to me, and at least does not correspond to the PR subject above.

If I'm confused, please explain. TIA.

**Review by TurboGit — COMMENTED** [review]
Patch is suspicious to me.

**Comment by parafin** [comment]
Also I’m not sure how we want to handle symlinks… It’s really good to have canonical path for the image to avoid trouble with duplicate imports, especially on Windows, where there’s so many ways to write it. On the other hand there is a use-case for not resolving symlinks. Symlinks aren’t used on Windows, so maybe make this call Windows-specific? Third point is difference between handling of absolute and relative paths.

In any case this stuff is fragile, so one first needs to understand why it was coded like it is now, what implications are of changing it.

**Comment by TurboGit** [comment]
Given the lack of support for this and that there is no discussion from OP I'll close.

**Comment by UweOhse** [comment]
My patch removes the call to g_realpath from the one code path using it (which is inconsistent), und creates an absolute path (i don't remember why, but seem to remember that darktable needs one) [you called the absolute path "full pathname"].

Which is what i said in the original PR - and i really do not know how to phrase it differently.

I'm opposed to using realpath, because that makes unusable at least one setup i used in the past:
   /mnt/readonlymedia/P123.RW2
   /home/work/P123.RW2 -> /mnt/readonlymedia/P123.RW2
That would fail as soon as /mnt/readonlymedia/P123.RW2.xmp would be created (with "cd /home/work; darktable P123.RW2"), and behave differently if "darktable  /home/work/P123.RW2" is called (/home/work/P123.RW2.xmp would be created).

I've got no opinion at all about the behavior on windows, as i've got no knowledge about the WIN32 API or the behavior of symlinks on windows. 

I rarely use github these days. It's unlikely i'll see answers / questions.

File: src/common/utility.c (hunk 2 above)
Comment (UweOhse) [reply src/common/utility.c:674]: The other code path, in which darktable got a "full path name", doesn't resolve symlinks, too.

Which is the inconsistency the OP in the original PR talked about.

**Pull request closed without merging**